packages/labs/virtualizer/src/test/helpers.ts

```diff
@@ -107,29 +107,67 @@ export function isInViewport(element: Element, viewport?: Element) {
 }
 
 /**
- * A promise which will resolve to the first truthy result of condition
- * function or the last result of calling it within the given timeout.
- * The intended usage of this function in a test would look something
- * like:
- *
- *     const thing = await eventually(() => doc.query('thing'));
- *
+ * Not exporting this function because the name is too generic and
+ * there are two more semantically meaningful use cases expressed by
+ * its dependent functions `pass` and `until`.
  */
-export async function eventually<T>(cond: () => T, timeout = 1000): Promise<T> {
+async function eventually<T>(cond: () => T, timeout = 1000): Promise<T> {
   const start = new Date().getTime();
-  return new Promise((resolve, _reject) => {
+  return new Promise((resolve, reject) => {
     check();
     function check() {
-      const result = cond();
-      if (result || new Date().getTime() - start > timeout) {
+      const [result, err] = testCond();
+      if (result) {
         return resolve(result);
       }
+      if (new Date().getTime() - start > timeout) {
+        if (err) {
+          return reject(err);
+        }
+        return resolve(result!);
+      }
       setTimeout(check, 0);
     }
+    function testCond(): [T?, Error?] {
+      try {
+        return [cond(), undefined];
+      } catch (e: unknown) {
+        return [undefined, e as Error];
+      }
+    }
   });
 }
 
 /**
+ * Returns a promise for the result of the code block given.  Errors are swallowed
+ * by the function until the timeout is reached, in which case the error will be
+ * thrown.  This means you can `pass` to wait for expectations like so:
```

Comment:
Maybe better wording here?
```suggestion
 * Returns a promise for the result of the code block given.  Errors are swallowed
 * by the function until the timeout is reached, after which the error will be
 * thrown.  This means you can use `pass` to wait for expectations like so:
```

Reply:
I also thought about if `pass` should just return `void` if the main use case is for retrying until an exception is not thrown.

Leaving as is does allow usage for callback to retrieve a value instead of just running assertions too. Semantically a bit further from `pass` though.

Reply:
Def better wording.  Applying.